api/v1alpha1/instancegroup_types.go

```diff
@@ -179,6 +179,10 @@ type EKSManagedSpec struct {
 	EKSManagedConfiguration *EKSManagedConfiguration `json:"configuration"`
 }
 
+type BootstrapOptions struct {
+	MaxPods int64 `json:"maxPods,omitempty"`
```

Comment:
are we planning to add a `KubeletArguments` element struct down the road to support all possible kubelet options in free form? if so how would we handle a `--max-pods` flag provided there directly?
e.g.
```
configuration:
  bootstrapOptions:
    kubeletArguments: "--max-pods 10 --eviction-hard memory.available<300mi"
    maxPods: 20
```
Since you are abstracting the UseMaxPods bool into just taking the maxPods (which is definitely nicer), we need to think of what to do in case of 'conflict' (this is only a problem in non-bottlerocket OS).

Also, what will we do in cases where we get both bootstrapArguments and `kubeletArguments` prior to deprecation of `bootstrapArguments`? e.g. 
```
configuration:
  bootstrapArguments: "--eviction-hard memory.available<600mi"
  bootstrapOptions:
    kubeletArguments: "--eviction-hard memory.available<300mi"
```

Reply:
Great questions!

Given that we don't have an API for BottleRocket kubelet args at the moment, I don't want to get too deep into designing an API for it. I think a freeform string for now works - it's likely that for bottlerocket, we could leverage the same field to just pass in TOML. 

I'm thinking that we just allow duplicate flags. I ran into this while testing, actually - Kubelet just uses the last argument passed in. With that in mind, I think we should go the following strategy:

```bash
--kubelet-extra-args '<bootstrapArguments> <kubeletArguments> <generatedArguments'
```

This will result in:
- Generated kubelet args having the highest priority
- The 'new' kubeletArguments having normal priority
- The to-be-deprecated bootstrapArguments having the lower priority.

Reply:
Awesome, this approach should be good !